rskj-core/src/main/java/co/rsk/RskContext.java

```diff
@@ -1330,6 +1330,12 @@ public synchronized void close() {
             logger.trace("wallet closed.");
         }
 
+        if (snapshotProcessor != null) {
```

Comment:
do we need this? I believe it's being disposed in the node runner

Reply:
We need to dispose the key value datasource when the node is closed

Reply:
but it already gets closed. SnapshotProcessor implements InternalService, which is being managed, as any other service, by the node runner.